ta/avb/entry.c

```diff
@@ -19,28 +23,78 @@ static TEE_Result get_slot_offset(size_t slot, size_t *offset)
 	return TEE_SUCCESS;
 }
 
-static TEE_Result create_state(uint32_t lock_state, TEE_ObjectHandle *h)
+static TEE_Result create_rb_state(uint32_t lock_state, TEE_ObjectHandle *h)
 {
 	const uint32_t flags = TEE_DATA_FLAG_ACCESS_READ |
 			       TEE_DATA_FLAG_ACCESS_WRITE |
 			       TEE_DATA_FLAG_OVERWRITE;
 
-	return TEE_CreatePersistentObject(storageid, obj_name, sizeof(obj_name),
-					  flags, NULL, &lock_state,
-					  sizeof(lock_state), h);
+	return TEE_CreatePersistentObject(storageid, rb_obj_name,
+					  sizeof(rb_obj_name), flags, NULL,
+					  &lock_state, sizeof(lock_state), h);
 }
 
-static TEE_Result open_state(uint32_t default_lock_state, TEE_ObjectHandle *h)
+static TEE_Result open_rb_state(uint32_t default_lock_state,
+				TEE_ObjectHandle *h)
 {
-	uint32_t flags = TEE_DATA_FLAG_ACCESS_READ | TEE_DATA_FLAG_ACCESS_WRITE;
+	uint32_t flags = TEE_DATA_FLAG_ACCESS_READ |
+			 TEE_DATA_FLAG_ACCESS_WRITE;
 	TEE_Result res;
 
-	res = TEE_OpenPersistentObject(storageid, obj_name,
-				       sizeof(obj_name), flags, h);
+	res = TEE_OpenPersistentObject(storageid, rb_obj_name,
+				       sizeof(rb_obj_name), flags, h);
+	if (!res)
+		return TEE_SUCCESS;
+
+	return create_rb_state(default_lock_state, h);
+}
+
+static TEE_Result create_named_value(char *obj_name, char *value,
+				     size_t value_size, TEE_ObjectHandle *h)
+{
+	const uint32_t flags = TEE_DATA_FLAG_ACCESS_READ |
+			       TEE_DATA_FLAG_ACCESS_WRITE |
+			       TEE_DATA_FLAG_OVERWRITE;
+
+	if (strlen(obj_name) >= TEE_OBJECT_ID_MAX_LEN)
+		return TEE_ERROR_ITEM_NOT_FOUND;
+
+	return TEE_CreatePersistentObject(storageid, obj_name,
+					  strlen(obj_name) + 1,
+					  flags, NULL, value,
+					  value_size, h);
+}
+
+static TEE_Result open_named_value(char *obj_name, uint32_t obj_max_size,
+				   char *value, size_t value_size,
+				   bool create, TEE_ObjectHandle *h)
+
+{
+	uint32_t flags = TEE_DATA_FLAG_ACCESS_READ |
+			 TEE_DATA_FLAG_ACCESS_WRITE;
+	TEE_Result res;
+	char name[TEE_OBJECT_ID_MAX_LEN];
+
+	if (strnlen(obj_name, obj_max_size) + strlen(named_value_prefix) >=
```

Comment:
It would be easier if the client supplied the length of the string in what's now `obj_max_size` instead of just supplying the upper limit (which I assume in practice is the same).

Reply:
In fact, it's `strlen(name) + 1;` (http://patchwork.ozlabs.org/patch/1013658/)

Reply:
So why not take advantage of that instead?

Reply:
ok. will fix

Reply:
fixed